devtools/create-gossipstore.c

```diff
@@ -4,24 +4,44 @@
 #include <ccan/read_write_all/read_write_all.h>
 #include <common/type_to_string.h>
 #include <common/utils.h>
+#include <devtools/create-gossipstore.h>
 #include <fcntl.h>
 #include <gossipd/gen_gossip_store.h>
 #include <gossipd/gossip_store.h>
 #include <inttypes.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
 #include <wire/gen_peer_wire.h>
 
+
+
+struct scidsat * load_scid_file(FILE * scidfd) 
+{
+        int n;
+        fscanf(scidfd, "%d\n", &n);	
```

Comment:
Generally, you should always validate input.  In this case, it means checking that fscanf returns 1.